main/src/cgeo/geocaching/filter/AbstractFilter.java

```diff
@@ -29,13 +29,19 @@ protected AbstractFilter(final Parcel in) {
 
     @Override
     public void filter(@NonNull final List<Geocache> list) {
-        final List<Geocache> itemsToRemove = new ArrayList<>();
+
+        //method must be performant when used with very large lists (e.g. 50000 elements) filtered into very short lists (e.g. 30 elements)
```

Comment:
Sorry for random comment, but shouldn't be this method performant in all the cases, including opposite one - filtering very large lists (50k) into a very large list (49k)?

Reply:
It is performant in all the cases 😉

Reply:
Just from the code, existing version must have been faster for my case. Current code is creating the copy-list with almost the same caches leading to some potential memory overhead and then clears and re-adds items again (so feels like ~3x overhead on the initial size of the list).
Previous version would have very short `itemsToRemove` (i.e. 1 element) and then iterate over the existing list once (for 1 element), shrinking it once... I'm not convinced new approach is faster in this case :)

Reply:
@okainav you are right, at least I should have tested this case too in unit tests, and also the comment is misleading.
In general I assume filtering large lists into large results should work fine performance-wise since addAll from arraylist to arraylist should be a single arraycopy underneath (performance-wise same thing happening with every single remove-call on such a list). Memory-wise I agree that temporary a second listsl of size 50000 has to be kept in memory, but this only adds 50000 pointers (not deep copies of the cache data) to existing 50000 complete caches stored in memory - so in relation it should not add much to the (huge!) Load of holding so many caches in memory anyway.

Reply:
@okainov I created PR #9090 for this